vllm/multimodal/cache.py

```diff
@@ -399,7 +520,10 @@ def processor_cache_from_config(
     if not _enable_ipc_cache(vllm_config):
         return MultiModalProcessorOnlyCache(model_config)
 
-    return MultiModalProcessorSenderCache(model_config)
+    if _enable_mm_input_shm_cache(vllm_config):
+        return ShmObjectStoreSenderCache(vllm_config)
+    else:
+        return MultiModalProcessorSenderCache(model_config)
```

Comment:
```suggestion
    if not _enable_mm_input_shm_cache(vllm_config):
        return MultiModalProcessorSenderCache(model_config)

    return ShmObjectStoreSenderCache(vllm_config)
```

Put the less preferred options at the top